### DISCUSSION TITLE
SQL语句合法性

### DISCUSSION THREAD

**Discussion by ghost** [liusy58/NJU_AdvancedProgramming_SP22#45] [Q&A] [answered]
proj不要求实现直接输入SQL语句的接口。那么是否可以假定，所有传入SQL模块的语句都是合法的SQL语句，且都是按照ppt中格式给出的（比如以单个空格分隔字符串，insert into的values以单个逗号分隔，update的多个values以单个逗号+单个空格分隔）？

**Comment by binbin00** [comment, marked as answer]
考虑到整个项目的工作量，我们不会对这个模块进行测试，只会查看指令日志文件及指令合成和解析的源代码。因此，传入SQL模块的语句都是你们自己生成的。

**Reply by ghost** [reply]
> 考虑到整个项目的工作量，我们不会对这个模块进行测试，只会查看指令日志文件及指令合成和解析的源代码。因此，传入SQL模块的语句都是你们自己生成的。

那仅仅通过判断语句首字母，来确定语句类型 这种行为是否被允许呢？

**Reply by Price1999a** [reply]
这属于你个人的实现细节 你觉得可以 那就可以。
